### PR TITLE
KAFKA-10367: Allow running the Streams demo app with a config file

### PR DESCRIPTION
**Pull request by mimaison**
Update the 3 WordCount demos to accept a configuration file. I kept the changes to the minimum as the point of these samples is not to add a lot of logic for parsing arguments and handling usage errors.

*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mjsax** [comment]
Checkstyle error:
```
> Task :streams:examples:checkstyleMain FAILED
02:46:32 [ant:checkstyle] [ERROR] /home/jenkins/jenkins-slave/workspace/kafka-pr-jdk11-scala2.13/streams/examples/src/main/java/org/apache/kafka/streams/examples/wordcount/WordCountDemo.java:52:49: Variable 'args' should be declared final. [FinalLocalVariable]
```

File: streams/examples/src/main/java/org/apache/kafka/streams/examples/wordcount/WordCountDemo.java [hunk 1]
```diff
@@ -47,18 +49,23 @@
...
 
-    static Properties getStreamsConfig() {
+    static Properties getStreamsConfig(final String[] args) throws IOException {
         final Properties props = new Properties();
-        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "streams-wordcount");
-        props.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
-        props.put(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, 0);
-        props.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
-        props.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
+        if (args != null && args.length == 1) {
```
Comment (mjsax) [inline streams/examples/src/main/java/org/apache/kafka/streams/examples/wordcount/WordCountDemo.java]: For `args.length > 1` should we log a warning that some parameters are ignored and still use the first parameter as file name? -- Just ignoring all arguments silently (if there is more than one) seems not ideal?

File: streams/examples/src/test/java/org/apache/kafka/streams/examples/wordcount/WordCountDemoTest.java [hunk 2]
```diff
@@ -47,11 +48,11 @@
     private TestOutputTopic<String, Long> outputTopic;
 
     @Before
-    public void setup() {
+    public void setup() throws IOException {
         final StreamsBuilder builder = new StreamsBuilder();
         //Create Actual Stream Processing pipeline
         WordCountDemo.createWordCountStream(builder);
-        testDriver = new TopologyTestDriver(builder.build(), WordCountDemo.getStreamsConfig());
+        testDriver = new TopologyTestDriver(builder.build(), WordCountDemo.getStreamsConfig(new String[] {}));
```
Comment (mjsax) [inline streams/examples/src/test/java/org/apache/kafka/streams/examples/wordcount/WordCountDemoTest.java]: Nit: We can just pass in `null` ?

File: streams/examples/src/main/java/org/apache/kafka/streams/examples/wordcount/WordCountDemo.java [hunk 3]
```diff
@@ -47,18 +49,23 @@
...
+            try (FileInputStream fis = new FileInputStream(args[0])) {
```
Comment (mjsax) [inline streams/examples/src/main/java/org/apache/kafka/streams/examples/wordcount/WordCountDemo.java]: I think this must be `final`, too

File: streams/examples/src/main/java/org/apache/kafka/streams/examples/wordcount/WordCountProcessorDemo.java [hunk 4]
```diff
@@ -101,16 +103,20 @@ public void close() {}
...
-    public static void main(final String[] args) {
+    public static void main(final String[] args) throws IOException {
         final Properties props = new Properties();
-        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "streams-wordcount-processor");
-        props.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
-        props.put(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, 0);
-        props.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass());
-        props.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass());
+        if (args != null && args.length == 1) {
+            props.load(new FileInputStream(args[0]));
```
Comment (mjsax) [inline streams/examples/src/main/java/org/apache/kafka/streams/examples/wordcount/WordCountProcessorDemo.java]: Missing `try` block to close the `FileInputStream` properly

File: streams/examples/src/main/java/org/apache/kafka/streams/examples/wordcount/WordCountTransformerDemo.java [hunk 5]
```diff
@@ -120,16 +122,19 @@ public void close() {}
...
-    public static void main(final String[] args) {
+    public static void main(final String[] args) throws IOException {
         final Properties props = new Properties();
-        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "streams-wordcount-transformer");
-        props.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
-        props.put(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, 0);
-        props.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass());
-        props.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass());
+        if (args != null && args.length == 1) {
+            props.load(new FileInputStream(args[0]));
```
Comment (mjsax) [inline streams/examples/src/main/java/org/apache/kafka/streams/examples/wordcount/WordCountTransformerDemo.java]: as above

**Review by mjsax — COMMENTED** [review]
Overall LGTM. Couple of comments.

**Comment by mimaison** [comment]
Thanks @mjsax, I wasn't expecting a review this quickly! Sorry for not fully checking the changes before opening the PR.

**Comment by mimaison** [comment]
@mjsax Can you take another look? Thanks

**Review by mjsax — APPROVED** [review]
Thanks @mimaison. LGTM.

**Pull request merged**